sources/PseudoTerminal.m

```diff
@@ -712,6 +735,10 @@ - (void)finishInitializationWithSmartLayout:(BOOL)smartLayout
             initialFrame = [screen visibleFrameIgnoringHiddenDock];
             break;
 
+        case WINDOW_TYPE_COMPACT_MAXIMIZED:
+        case WINDOW_TYPE_MAXIMIZED:
+            initialFrame = [screen visibleFrame];
```

Comment:
Hmm, so it turns out I had missed adding a "break" on this case statement. That caused initialFrame to actually be wrong initially. But oddly, fixing that so that initialFrame is correct causes some weird layout glitch (an empty 22px space at the top of the window below the title bar), which I've managed to trace to addShortcutAccessorViewControllerToTitleBarIfNeeded. I'm adding a weird workaround for it with a comment explaining...